src/components/HeaderContainer.tsx

```diff
@@ -0,0 +1,16 @@
+import { RefObject } from 'react';
+import HeaderUI from './HeaderUI';
+
+interface HeaderContainerProps {
+  deleteModalRef: RefObject<HTMLDialogElement>;
+}
+
+function HeaderContainer({ deleteModalRef }: HeaderContainerProps) {
+  const openDeleteModal = () => {
```

Comment:
Question 🤔: why can't we just pass in the openDeleteModal callback instead of the ref?

Reply:
updated to define `openDeleteModal` in `App.tsx` and removed `HeaderContainer.tsx` as it was not handling any logic anymore. `openDeleteModal` now passed as prop to `HeaderUI.tsx`